VirtualApp/app/build.gradle

```diff
@@ -22,12 +22,12 @@ android {
         }
     }
 
-    compileSdkVersion 26
-    buildToolsVersion '26.0.2'
+    compileSdkVersion 27
+    buildToolsVersion '27.1.1'
     defaultConfig {
         applicationId "io.va.exposed"
         minSdkVersion 21
-        targetSdkVersion 23
+        targetSdkVersion 27
```

Comment:
targetSdkVersion may affect the behavior of VirtualApp, the version 23 is picked out carefully, it shouldn't be changed now.

Reply:
hmm

Reply:
ill see how to work with this tomorrow

Reply:
so VirtualApp needs to be updated with these for this to be able to work

Reply:
upgrade the targetSdkVersion has lots of work to do, and it is unnecessary.

Reply:
thats partly because VirtualApp itself hasnt been updated with these, i think